### DISCUSSION TITLE
Nearest boundary node from every other node in a mesh

### DISCUSSION THREAD

**Discussion by Eilloo** [idaholab/moose#22855] [Q&A General] [answered]
Hello all,

I've been trying to make use of the 'NearestNodeDistanceAux' and 'NearestNodeValuesAux' auxkernels, but have only managed to find the nearest nodes on one boundary from another.

As the question title suggests, I'd like to extract information about the nearest node on a certain boundary to _every_ node in the mesh, not just those which are on another specific boundary.

I believe the documentation suggests this is possible; however, if I only specify the 'paired boundary' (required parameter), but do not specify any 'boundary' parameter (optional parameters), the error reads:

'"Please use getBoundaryIDs() when passing "ANY_BOUNDARY_ID"'

I thought that specifying the 'block_id' instead of a boundary might be the intended usage, but this also yields the error above.

Does anyone have experience with this auxkernel who can advise how to use it to get the nearest boundary node from every point, or if this can be achieved at all?

Thanks!

**Comment by maxnezdyur** [comment, marked as answer]
I have no experience with the auxkernel, so there may be a better MOOSE way to get the functionality you want. If your geometry is simple, you may be able to use `BoundingBoxNodeSetGenerator` in the mesh block to create a boundary that holds all the nodes that you care about within it and supply that as the `boundary` parameter.

**Comment by GiudGiud** [comment]
This object was developed to look at contact between blocks. And logically, the nearest node to a boundary on a part of the mesh is on the boundary of that other part of the mesh, it cannot be inside the volume. 

Can you create a boundary around the domain you are trying to find the nearest node from? Logically you only really need to look at points on a boundary.

If not, would you consider uploading a sketch or a figure of the domains you are trying to search for the nearest node?

**Comment by Eilloo** [comment]
Thanks for the suggestions and insight - @maxnezdyur, this worked a treat.

@GiudGiud, that makes sense why the object operates the way it does. In my case, the ability to easily get the distance from any node to a wall boundary is useful for looking at y plus and wall functions (related to my question on the Smagorinsky kernel which you helped answer a couple of weeks ago!).

I should think that even for complex geometry, using BoundingBoxNodeSetGenerator around the whole domain should enable me to use the 'NearestNode' auxkernels in the way I was hoping.

Thanks again!
